hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/UserAgentSpanEnricher.java

```diff
@@ -59,4 +54,21 @@ public void enrichEvent(StructuredTrace trace, Event event) {
           AttributeValueCreator.create(userAgent.getVersionNumber().toVersionString()));
     }
   }
+
+  private Optional<String> getUserAgent(Event event) {
+    if (event.getHttp() != null && event.getHttp().getRequest() != null) {
+      // prefer user agent from headers
+      if (event.getHttp().getRequest().getHeaders() != null
+          && !StringUtils.isEmpty(event.getHttp().getRequest().getHeaders().getUserAgent())) {
+        return Optional.of(event.getHttp().getRequest().getHeaders().getUserAgent());
+      }
+
+      // fallback to user agent on the request
+      if (!StringUtils.isEmpty(event.getHttp().getRequest().getUserAgent())) {
```

Comment:
Huh. Weird. Can you check if the one on the request is a copy of the header one? Could be. Also we should make changes to stop reading from enriched attributes for the consumers of this. This can just read this and there would be no need to enrich.

Reply:
I checked. The user agent coming from request header gets set in `getHttp().getRequest().getHeaders().getUserAgent()`, whereas the others gets set in `getHttp().getRequest().getUserAgent()`

https://github.com/hypertrace/span-normalizer/blob/main/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/fieldgenerators/HttpFieldsGenerator.java#L258

Reply:
The consumers would have to parse the user agent and get the name, type, device, category, which this enricher is doing for them

So, I think it makes sense to enrich the derived values from user agent string in the enricher